samples/CallWithChat/src/app/views/CallScreen.tsx

```diff
@@ -30,37 +30,16 @@ export interface CallScreenProps {
 
 export const CallScreen = (props: CallScreenProps): JSX.Element => {
   const { token, userId, displayName, endpoint, locator } = props;
-  const [adapter, setAdapter] = useState<CallWithChatAdapter>();
   const callIdRef = useRef<string>();
-  const adapterRef = useRef<CallWithChatAdapter>();
   const { currentTheme, currentRtl } = useSwitchableFluentTheme();
-  const [isMobileSession, setIsMobileSession] = useState<boolean>(detectMobileSession());
 
-  // Whenever the sample is changed from desktop -> mobile using the emulator, make sure we update the formFactor.
-  useEffect(() => {
-    const updateIsMobile = (): void => {
-      // The userAgent string is sometimes not updated synchronously when the `resize` event fires.
-      setTimeout(() => {
-        setIsMobileSession(detectMobileSession());
-      });
-    };
-    window.addEventListener('resize', updateIsMobile);
-    return () => window.removeEventListener('resize', updateIsMobile);
-  }, []);
-
-  useEffect(() => {
-    (async () => {
-      if (!userId || !displayName || !locator || !token || !endpoint) {
-        return;
-      }
+  const credential = useMemo(
+    () => createAutoRefreshingCredential(toFlatCommunicationIdentifier(userId), token),
+    [userId, token]
+  );
 
-      const adapter = await createAzureCommunicationCallWithChatAdapter({
-        userId,
-        displayName,
-        credential: createAutoRefreshingCredential(toFlatCommunicationIdentifier(userId), token),
-        endpoint,
-        locator: locator
-      });
+  const afterAdapterCreate = useCallback(
```

Comment:
This looks good - I am thinking of what is the best practice to use afterAdapterCreate vs useAzureCommunicationCallWithChatAdapter + useEffect

Imagine a piece of code like this, it seems to do the same job:

```
const adapter = useAdapter(...);
useEffect(()=> {
  if(adapter) {
    adapter.on('error', (e) => {...});
    decoratedAdapter = decorateAdapter(adapter);
  }
}, [adapter]);
```

Is there a specific case we want to ask developers to use afterAdapterCreate over useAzureCommunicationCallWithChatAdapter + useEffect?

Reply:
Would be nice to document this if there is any preference

Reply:
`afterCreate` is a direct replacement for the extra `useEffect`. Less is more, so use this not `useEffect` :)

As for documentation, we use this in our samples after this PR, and we will also use `beforeDispose` in storybook for `CallAdapter` (follow up PR).